purchase_stock_bl_date/models/purchase.py

```diff
@@ -0,0 +1,39 @@
+# Copyright 2023 Quartile Limited
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import api, fields, models
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    bl_date = fields.Date()
+
+    @api.multi
+    def action_view_invoice(self):
+        res = super(PurchaseOrder, self).action_view_invoice()
+        if res.get("context"):
+            res["context"].update(
+                default_date_invoice=self.bl_date,
+            )
+        return res
+
+    @api.multi
+    def _create_picking(self):
+        res = super(PurchaseOrder, self)._create_picking()
+        for order in self:
+            if order.picking_ids:
+                order.picking_ids.write({"bl_date": order.bl_date})
+        return res
+
+
+class PurchaseOrderLine(models.Model):
+    _inherit = "purchase.order.line"
+
+    @api.multi
+    def _get_stock_move_price_unit(self):
+        self.ensure_one()
+        bl_date = self.order_id.bl_date
+        if bl_date:
+            self = self.with_context(bl_date=bl_date)
+        return super()._get_stock_move_price_unit()
```

Comment:
IMO, we should remove this (unless I'm overlooking something).  There is be no difference whether B/L date is available or not - when an incoming move is created from a PO, we just follow the standard logic.  What matters is the correct application of the unit_price where it affects the stock valuation or journal entries.  Everything else does not matter.

```suggestion
```